eth/catalyst/api.go

```diff
@@ -335,78 +191,40 @@ func (api *ConsensusAPI) insertTransactions(txs types.Transactions) error {
 
 func (api *ConsensusAPI) checkTerminalTotalDifficulty(head common.Hash) error {
 	// shortcut if we entered PoS already
-	if api.merger().PoSFinalized() {
-		return nil
-	}
-	if api.light {
-		// make sure the parent has enough terminal total difficulty
-		header := api.les.BlockChain().GetHeaderByHash(head)
-		if header == nil {
-			return &GenericServerError
-		}
-		td := api.les.BlockChain().GetTd(header.Hash(), header.Number.Uint64())
-		if td != nil && td.Cmp(api.les.BlockChain().Config().TerminalTotalDifficulty) < 0 {
-			return &InvalidTB
-		}
+	if api.eth.Merger().PoSFinalized() {
 		return nil
 	}
 	// make sure the parent has enough terminal total difficulty
 	newHeadBlock := api.eth.BlockChain().GetBlockByHash(head)
 	if newHeadBlock == nil {
-		return &GenericServerError
+		return &beacon.GenericServerError
 	}
 	td := api.eth.BlockChain().GetTd(newHeadBlock.Hash(), newHeadBlock.NumberU64())
 	if td != nil && td.Cmp(api.eth.BlockChain().Config().TerminalTotalDifficulty) < 0 {
-		return &InvalidTB
+		return &beacon.InvalidTB
```

Comment:
@MariusVanDerWijden 

What if the `td` is nil while the headBlock is present? In case of some database corruption?

Reply:
This should only happen if the database is fried and we have the headBlock in DB, but not the TD for it in the DB.
I'm not sure if we should handle it here or if this should blow up at some other point.
Maybe @karalabe can weigh in here: Should we double check and error or should we ignore and eventually blow up in case of database corruption

Reply:
What about returning a `&beacon.GenericServerError` here?

Reply:
Yeah, might be better than silently accepting